### PR TITLE
[BEAM-1127] Create an unique source when using a JMS topic to avoid p…

### PR DESCRIPTION
**Pull request by jbonofre**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

…otential messages duplication

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @dhalperi 
CC: @davorbonaci 
CC: @jkff

File: sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java [hunk 1]
```diff
@@ -236,8 +236,15 @@ public UnboundedJmsSource(
     public List<UnboundedJmsSource> generateInitialSplits(
         int desiredNumSplits, PipelineOptions options) throws Exception {
       List<UnboundedJmsSource> sources = new ArrayList<>();
-      for (int i = 0; i < desiredNumSplits; i++) {
+      if (topic != null) {
+        // in the case of a topic, we create a single source, so an unique subscriber, to avoid
```
Comment (davorbonaci) [inline sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java]: I might be totally uninformed, but is it possible to make a queue out of a topic? Said differently, is it possible to separate a given topic into multiple queues?

File: sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java [hunk 2]
```diff
@@ -236,8 +236,15 @@ public UnboundedJmsSource(
     public List<UnboundedJmsSource> generateInitialSplits(
         int desiredNumSplits, PipelineOptions options) throws Exception {
       List<UnboundedJmsSource> sources = new ArrayList<>();
-      for (int i = 0; i < desiredNumSplits; i++) {
+      if (topic != null) {
```
Comment (davorbonaci) [inline sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java]: Is it reasonable to add a unit test that fails before this change and passes after this change, just to make sure we don't regress in the future?

**Comment by dhalperi** [comment]
Concur that this behavior should be tested.

LGTM otherwise.

File: sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java (hunk 1 above)
Comment (jbonofre) [inline sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java]: It's possible but on the broker side. In ActiveMQ, it's what a Virtual Topic.

**Comment by jbonofre** [comment]
Gonna rebase and add a test to check the number of source created is correct depending of the destination (topic or queue). Thanks !

**Comment by jbonofre** [comment]
Rebased and added a test to illustrate the expected behavior.

**Comment by dhalperi** [comment]
LGTM, thanks JB!

**Pull request closed without merging**